kvdb-memorydb/src/lib.rs

```diff
@@ -44,34 +48,34 @@ pub fn create(num_cols: u32) -> InMemory {
 	}
 }
 
-impl KeyValueDB for InMemory {
-	fn get(&self, col: Option<u32>, key: &[u8]) -> io::Result<Option<DBValue>> {
+impl KeyValueDB<DBKey, DBValue> for InMemory {
+	fn get(&self, col: Option<u32>, key: &DBKey) -> io::Result<Option<DBValue>> {
 		let columns = self.columns.read();
 		match columns.get(&col) {
 			None => Err(io::Error::new(io::ErrorKind::Other, format!("No such column family: {:?}", col))),
 			Some(map) => Ok(map.get(key).cloned()),
 		}
 	}
 
-	fn get_by_prefix(&self, col: Option<u32>, prefix: &[u8]) -> Option<Box<[u8]>> {
+	fn get_by_prefix(&self, col: Option<u32>, prefix: &[u8]) -> Option<DBValue> {
 		let columns = self.columns.read();
 		match columns.get(&col) {
 			None => None,
-			Some(map) =>
-				map.iter()
-					.find(|&(ref k ,_)| k.starts_with(prefix))
-					.map(|(_, v)| v.to_vec().into_boxed_slice())
+			Some(map) => map.iter()
+				.find(|&(ref k ,_)| k.starts_with(prefix))
+				.map(|(_, v)| v)
+				.cloned(),
 		}
```

Comment:
Not introduced by this PR but pattern matching could be replaced by `and_then` to eliminate `None => None` etc

```rust
		self.columns.read().get(&col)
			.and_then(|db| {
				db.iter()
					.find(|&(ref k ,_)| k.starts_with(prefix))
					.map(|(_, v)| v)
					.cloned()
			})

```